### ISSUE TITLE
Randomly failing SignalR connections

### ISSUE THREAD

**Issue by Stamo-Gochev** [aspnet/AspNetCore#6797]
### Describe the bug
I am experiencing a strange problem with failing connections to a SignalR service, which seem to be happening from time to time (refreshing the page multiple times results in several successful and failing requests). I tested with different loads of the machine where the service is deployed and this doesn't seem to affect the issue - some requests are failing with basically no load at all. 

The exact problem is failing GET requests when Long polling transport is used and they break with 404 Not Found when hitting a URL like:
```
https://<some-url>?id=UaxnI4GYMOyMjtMW6y_E3w&_=1547715317314 
```

### To Reproduce
The problem can be observed by opening the following page:

https://jsbin.com/suqabipiyi/1/edit?html,output

It connects to the service, calls a ["read" method](https://github.com/telerik/kendo-ui-demos-service/blob/master/signalr-hubs/signalr-for-aspnet-core/Hubs/ProductHub.cs#L24) and displays the data. Note that although the result is returned, the connection is lost and no further actions can be performed. You might need to run the example several times to observe the problem as it appears to happen randomly.

Here are the steps that can be used to test the project locally:
1. Use the project from the GitHub repo: https://github.com/telerik/kendo-ui-demos-service/tree/master/signalr-hubs
2. Publish the app with the standard instructions for publishing (or use the instructions from the repo).
3. Request the data as done in https://jsbin.com/suqabipiyi/1/edit?html,output

The issue is happening when Long polling is used. Doesn't happen with Server-sent events. 

### Expected behavior
Connections should not be failing randomly.

### Additional context

The result of `dotnet --info`: https://pastebin.com/PTriewcL
**SignalR version:** [2.1](https://github.com/telerik/kendo-ui-demos-service/blob/master/signalr-hubs/signalr-for-aspnet-core/signalr-for-aspnet-core.csproj#L8)
**SignalR script version:** [1.1.0](https://unpkg.com/@aspnet/signalr@1.1.0/dist/browser/signalr.js)

**Comment by mikaelm12** [comment]
Can you give us some more information about your configuration? Are you using a free tier of the SignalR service, or the standard level with multiple units? You're not seeing this issue with Websockets or SSE? I also wasn't able to repro your issue with the jsbin link you sent.

**Comment by mikaelm12** [comment]
My first thought since you're saying you get the issue seemingly randomly is that for whatever reason your cookies aren't being respected and that means stick sessions which are required aren't working on the Azure SignalR Service. This could cause seemingly random failures. But I can't be sure without more information.

**Comment by Stamo-Gochev** [comment]
1. The service is not hosted on Azure. More information about the deployment environment can be found in the result of the `dotnet --info` command from my post - https://pastebin.com/PTriewcL
2. The problem can be reproduced with Long polling, but **not** with Server-sent events.
3. In order to reproduce the issue, navigate to the link https://jsbin.com/suqabipiyi/1/edit?html,output and wait till the data is rendered on the page. Note that this might require refreshing the page as this happens randomly. If you are still not able to reproduce it, can you try by setting up the service locally as I described in the "Steps to reproduce" section? What is strange is that a result is returned (the rendered data) and the connection is closed after that.

Here is a log of the problems I see in the DevTools console when running the jsbin sample:https://pastebin.com/cnfWi6t0 and a screenshot https://ibb.co/2kYCf3Y

**Comment by Stamo-Gochev** [comment]
@mikaelm12 I've done some more testing and the issue can be reproduced with SSE and WebSockets transports but less frequently. Can you think of any other reason apart from load/latency? Any ideas how to further investigate the reason will also be helpful.

**Comment by mikaelm12** [comment]
Do you have sticky sessions enabled in your cloud provider?

**Comment by mikaelm12** [comment]
That's assuming you have multiple machines. Not having sticky sessions enabled could explain the 404

**Comment by Stamo-Gochev** [comment]
I am testing on a setup with a load balancer, but with one machine only in order to isolate the case. Sticky sessions are enabled. Can you reproduce the problem by locally deploying the service?

**Comment by bradygaster** [comment]
When we look at your log, we noticed that the payload is **large**. Is this screen shot a good indicator of the size of the payload you're sending on the connections that are intermittently closing? 

![image](https://user-images.githubusercontent.com/41929050/51702830-aba33c00-1fc9-11e9-9ee6-24d68d90aa88.png)

**Comment by BrennanConroy** [comment]
>Connections should not be failing randomly.

I was able to repro the issue with your jsbin.com example. The reason the connection fails is that it takes longer than 5 seconds for your longpolling connection to re-poll and so the server closes the connection. This could be caused by your server being overloaded and taking too long to accept new requests or the internet connection being slow. I reproed it by opening 100 connections in the browser at the same time, which either caused the browser to be slow or overloaded your server.
